### PR TITLE
Http log range v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/2485

Describe changes:
- Logs http content-range header value (in responses) in json output

Follows #3773 
- Parses content-range so as to log integer values for start, end and size
- Puts these data in the file info record (separate commit)

### CODE REVIEW

**Comment by victorjulien** [comment]
When adding a PR like this one, please add a link to the libhtp PR it depends on.

File: src/app-layer-htp-file.c [hunk 1]
```diff
@@ -150,6 +150,40 @@ int HTPFileOpen(HtpState *s, const uint8_t *filename, uint16_t filename_len,
...
+    SCEnter();
+
+    int retval;
+    FileContainer *files;
+
+    if (s == NULL) {
+        SCReturnInt(-1);
+    }
+
+    files = s->files_tc;
```
Comment (victorjulien) [inline src/app-layer-htp-file.c]: declare files here

File: src/app-layer-htp-file.c [hunk 2]
```diff
@@ -150,6 +150,40 @@ int HTPFileOpen(HtpState *s, const uint8_t *filename, uint16_t filename_len,
...
+    if (files == NULL) {
+        SCLogDebug("no files in state");
+        SCReturnInt(-1);
+    }
+
+    retval = FileSetRange(files, start, end);
```
Comment (victorjulien) [inline src/app-layer-htp-file.c]: declare retval here

File: src/output-json-http.c [hunk 3]
```diff
@@ -261,6 +261,26 @@ static void JsonHttpLogJSONBasic(json_t *js, htp_tx_t *tx)
...
+            const size_t size = bstr_len(h_content_range->value) * 2 + 1;
+            char string[size];
+            BytesToStringBuffer(bstr_ptr(h_content_range->value), bstr_len(h_content_range->value), string, size);
+            json_t *crjs = json_object();
+            if (crjs != NULL) {
+                json_object_set_new(crjs, "raw", SCJsonString(string));
+                htp_content_range_t crparsed;
+                if (htp_parse_content_range(h_content_range->value, &crparsed) == HTP_OK) {
+                    if (crparsed.start >= 0)
+                    json_object_set_new(crjs, "start", json_integer(crparsed.start));
```
Comment (victorjulien) [inline src/output-json-http.c]: indent

File: src/output-json-http.c [hunk 4]
```diff
@@ -261,6 +261,26 @@ static void JsonHttpLogJSONBasic(json_t *js, htp_tx_t *tx)
...
+                    if (crparsed.start >= 0)
+                    json_object_set_new(crjs, "end", json_integer(crparsed.end));
+                    if (crparsed.start >= 0)
+                    json_object_set_new(crjs, "size", json_integer(crparsed.size));
+                }
+                json_object_set_new(js, "http_content_range", crjs);
```
Comment (victorjulien) [inline src/output-json-http.c]: we can drop the 'http_' here. We already regret adding that to some of the other fields.

File: src/app-layer-htp-file.c (hunk 1 above)
Comment (catenacyber) [reply src/app-layer-htp-file.c]: ok

File: src/app-layer-htp-file.c (hunk 2 above)
Comment (catenacyber) [reply src/app-layer-htp-file.c]: ok

File: src/output-json-http.c (hunk 3 above)
Comment (catenacyber) [reply src/output-json-http.c]: well seen

File: src/output-json-http.c (hunk 4 above)
Comment (catenacyber) [reply src/output-json-http.c]: ok :)

**Comment by victorjulien** [comment]
Replaced by #3805

**Pull request closed without merging**